src/main/java/net/sf/jabref/gui/EntryTypeDialog.java

```diff
@@ -148,6 +174,95 @@ private JPanel createEntryGroupPanel(String groupTitle, Collection<EntryType> en
         return panel;
     }
 
+    private JPanel createIdFetcher(String groupTitle) {
+        JButton generateButton = new JButton(Localization.lang("Generate"));
+        JButton cancelButton = new JButton(Localization.lang("Cancel"));
+        JTextField idTextField = new JTextField("");
+        JComboBox<String> comboBox = new JComboBox<>();
+        getIdFetchers().forEach(n -> comboBox.addItem(n.getName()));
+        JLabel fetcherLabel = new JLabel("ID type"), idLabel = new JLabel("ID");
+
+        SwingWorker<Optional<BibEntry>, Void> fetcherWorker = new SwingWorker<Optional<BibEntry>, Void>() {
+            Optional<BibEntry> bibEntry = Optional.empty();
+            IdBasedFetcher fetcher;
+            String searchID;
+
+            @Override
+            protected Optional<BibEntry> doInBackground() throws Exception {
+                searchID = idTextField.getText();
+                if (!searchID.isEmpty()) {
+                    fetcher = EntryFetchers.getIdFetchers().get(comboBox.getSelectedIndex());
+                    try {
+                        bibEntry = fetcher.performSearchById(searchID);
+                        dispose();
+                    } catch (FetcherException e) {
+                        LOGGER.error("Error fetching from " + fetcher.getName(), e);
+                        JOptionPane.showMessageDialog(null, Localization.lang("Error_while_fetching_from_%0", fetcher.getName()), Localization.lang("Error_while_fetching"), JOptionPane.ERROR_MESSAGE);
+                    }
+                }
+                return bibEntry;
+            }
+
+            @Override
+            protected void done() {
+                if (isCancelled()) {
+                    JOptionPane.showMessageDialog(null, "Fetching was canceled", "Canceled fetching", JOptionPane.INFORMATION_MESSAGE);
+                    return;
+                }
+
+                try {
+                    Optional<BibEntry> result = get();
+                    if (result.isPresent()) {
+                        frame.getCurrentBasePanel().insertEntry(result.get());
+                    } else {
+                        JOptionPane.showMessageDialog(null, Localization.lang("No_entry_with_id_'%0'_for_fetcher_'%1'_was_found.", searchID, fetcher.getName()), Localization.lang("No_files_found."), JOptionPane.WARNING_MESSAGE);
```

Comment:
fetcher could be null I think (select a fetcher in the dialog but let the id empty -> click search)